src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java

```diff
@@ -361,12 +364,35 @@ private void verifyIsCoursePresent(String courseId) throws EntityDoesNotExistExc
         }
     }
 
-    private void verifyIsInstructorOfCourse(String courseId, String email) throws EntityDoesNotExistException {
-        InstructorAttributes instructor = instructorsLogic.getInstructorForEmail(courseId, email);
-        if (instructor == null) {
-            throw new EntityDoesNotExistException("User " + email + " is not a registered instructor for course "
-                                                + courseId + ".");
+    private void verifyIsUserOfCourse(String courseId, String giverEmail, String giverRole)
+            throws EntityDoesNotExistException {
+        if (Const.STUDENT.equals(giverRole)) {
+            StudentAttributes student = studentsLogic.getStudentForEmail(courseId, giverEmail);
+            if (student == null) {
+                throw new EntityDoesNotExistException("User " + giverEmail + " is not a registered student for course "
+                        + courseId + ".");
+            }
+        } else if (Const.INSTRUCTOR.equals(giverRole)) {
+            InstructorAttributes instructor = instructorsLogic.getInstructorForEmail(courseId, giverEmail);
+            if (instructor == null) {
+                throw new EntityDoesNotExistException("User " + giverEmail
+                        + " is not a registered instructor for course " + courseId + ".");
+            }
+        } else {
```

Comment:
use "else if" to check whether the `giverRole` is "team"or something similar, and "else" to throw an error if the giver is not of these three type.